File: arrays/csharp/ArrayExamples.cs
// Creating arrays
int[] arr1 = new int[0]; // Empty array - fixed size, O(1)
int[] arr2 = new int[10]; // Predefined size, default elements 0
int[] arr3 = { 1, 2, 3, 4, 5 }; // Initialized with values

Console.WriteLine(string.Join(", ", arr1));
Console.WriteLine(string.Join(", ", arr2));
Console.WriteLine(string.Join(", ", arr3));

// Using List<T> for dynamic operations
List<int> list = new List<int> { 1, 2, 3, 4 };

list.Add(5); // Add to end - O(1) amortized
list.Add(6);
Console.WriteLine(string.Join(", ", list));

list.Insert(0, 12); // Insert at start - O(n) due to shifting elements
Console.WriteLine(string.Join(", ", list));

list.Insert(2, 5); // Insert at arbitrary index - O(n)
Console.WriteLine(string.Join(", ", list));

// Accessing elements
int[] accessArr = { 10, 20, 30, 40 };
int value = accessArr[2]; // O(1)
Console.WriteLine(value);

// Traversing arrays
for (int i = 0; i < accessArr.Length; i++)
{ // O(n)
  Console.WriteLine(accessArr[i]);
}

foreach (int num in accessArr)
{ // O(n)
  Console.WriteLine(num);
}

// Traversing List
foreach (int num in list)
{ // O(n)
  Console.WriteLine(num);
}

File: arrays/java/ArrayExamples.java

import java.util.ArrayList;
import java.util.Arrays;

public class ArrayExamples {

    public static void main(String[] args) {
        // Creating arrays
        int[] arr1 = new int[0]; // Empty array - fixed size, O(1)
        int[] arr2 = new int[10]; // Predefined size, all elements default to 0
        int[] arr3 = {1, 2, 3, 4, 5}; // Initialized with values

        System.out.println(Arrays.toString(arr1));
        System.out.println(Arrays.toString(arr2));
        System.out.println(Arrays.toString(arr3));

        // Using ArrayList for dynamic operations
        ArrayList<Integer> list = new ArrayList<>(Arrays.asList(1, 2, 3, 4));

        list.add(5); // Add to end - O(1) amortized
        list.add(6);
        System.out.println(list);

        list.add(0, 12); // Insert at start - O(n) because elements are shifted
        System.out.println(list);

        list.add(2, 5); // Insert at arbitrary index - O(n) shifting elements
        System.out.println(list);

        // Accessing elements
        int[] accessArr = {10, 20, 30, 40};
        int value = accessArr[2]; // O(1)
        System.out.println(value);

        // Traversing arrays
        for (int i = 0; i < accessArr.length; i++) { // O(n)
            System.out.println(accessArr[i]);
        }

        for (int num : accessArr) { // Enhanced for-loop, O(n)
            System.out.println(num);
        }

        // Traversing ArrayList
        for (int num : list) { // O(n)
            System.out.println(num);
        }
    }
}
